### PR TITLE
chore: improve error catching in NodeBindings

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

This PR improves error handling during Node.js Environment creation. Previously, it was very hard to diagnose Environment creation failures owing to the fact that we only surfaced errors when creation failed in the renderer process, and even then only logged an error amounting to "this failed". This makes it much harder during upgrades, for example, to more effectively identify and fix bootstrapping problems with descriptive errors.

Before: 
```
[72201:0210/194105.829906:FATAL:node_bindings.cc(485)] Check failed: env.
```

After:
```
[75744:0210/194903.608995:ERROR:node_bindings.cc(500)] Failed to initialize node environment in process: browser with error: Missing internal module 'internal/process/promises'
[75744:0210/194903.609014:FATAL:node_bindings.cc(503)] Check failed: env.
```

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: shell/common/node_bindings.cc [hunk 1]
```diff
@@ -476,25 +475,33 @@ node::Environment* NodeBindings::CreateEnvironment(
...
+  if (try_catch.HasCaught()) {
+    std::string err_msg =
+        "Failed to initialize node environment in process: " + process_type;
+    v8::Local<v8::Value> exception = try_catch.Exception();
+    if (!exception->IsNull() && exception->IsObject()) {
+      v8::Local<v8::Value> error_message = try_catch.Exception();
+      v8::MaybeLocal<v8::Value> maybe_message = exception.As<v8::Object>()->Get(
+          context, gin::ConvertToV8(isolate, "message"));
+      std::string msg;
+      if (maybe_message.ToLocal(&error_message) && error_message->IsString()) {
```
Comment (RaisinTen) [inline shell/common/node_bindings.cc:494]:
```suggestion
      v8::MaybeLocal<v8::Value> maybe_message = exception.As<v8::Object>()->Get(
          context, gin::ConvertToV8(isolate, "message"));
      v8::Local<v8::Value> error_message;
      if (maybe_message.ToLocal(&error_message) && error_message->IsString()) {
        std::string msg;
```
* there's no need to assign `try_catch.Exception()` to the `error_message` local first if we're just going to change its value to the ToLocal of `maybe_message` right afterwards
* moved the `msg` string closer to where it is actually required

File: shell/common/node_bindings.cc [hunk 2]
```diff
@@ -476,25 +475,33 @@ node::Environment* NodeBindings::CreateEnvironment(
...
+    v8::Local<v8::Value> exception = try_catch.Exception();
+    if (!exception->IsNull() && exception->IsObject()) {
+      v8::MaybeLocal<v8::Value> maybe_message = exception.As<v8::Object>()->Get(
+          context, gin::ConvertToV8(isolate, "message"));
+      v8::Local<v8::Value> error_message;
+      if (maybe_message.ToLocal(&error_message) && error_message->IsString()) {
+        std::string msg;
+        if (gin::ConvertFromV8(isolate, error_message, &msg))
+          err_msg += " with error: " + msg;
+      }
```
Comment (deepak1556) [inline shell/common/node_bindings.cc:497]:
```suggestion
   v8::Local<v8::Message> message = try_catch.Message();
   std::string msg;
   if (!message.IsEmpty() && gin::ConvertFromV8(isolate, message->Get(), &msg))
       err_msg += " , with error: " + msg;
```

**Review by deepak1556 — COMMENTED** [review]
LGTM

File: shell/common/node_bindings.cc (hunk 2 above)
Comment (codebytere) [reply shell/common/node_bindings.cc]: @deepak1556 i went back to this and i do specifically want the message body - `message->Get()` returns the whole of `Uncaught Error: error` when i only want `error`, which is why i went with the previous approach. Are you ok with this or would you prefer to keep the `Uncaught Error` portion?

Comment (deepak1556) [reply shell/common/node_bindings.cc]: I think the retrieving the whole message is fine, also saves extra checks and conversions to v8 in the previous approach, whereas the current version gets the object already available in c++ and only requires a single conversion from v8.

**Pull request merged**